turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -794,22 +792,30 @@ impl<B: Backend + 'static> TurboTasks<B> {
         let future = async move {
             let TaskExecutionSpec { future, span } =
                 crate::task::local_task::get_local_task_execution_spec(&*this, &ty, persistence);
+            let ty = ty.clone();
             async move {
-                let (result, _duration, _memory_usage) =
-                    CaptureFuture::new(AssertUnwindSafe(future).catch_unwind()).await;
-
-                let result = result.map_err(|any| match any.downcast::<String>() {
-                    Ok(owned) => Some(Cow::Owned(*owned)),
-                    Err(any) => match any.downcast::<&'static str>() {
-                        Ok(str) => Some(Cow::Borrowed(*str)),
-                        Err(_) => None,
-                    },
-                });
+                let (result, _duration, _memory_usage) = CaptureFuture::new(future).await;
```

Comment:
Hmm... Looking back at this code, I'm wondering if I should make a lightweight version of `CaptureFuture` for local tasks that doesn't bother capturing duration or memory usage...